// @aws-sdk/client-lex-runtime-v2/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextInputEvent$ = exports.StartConversationResponse$ = exports.StartConversationRequest$ = exports.Slot$ = exports.SessionState$ = exports.SentimentScore$ = exports.SentimentResponse$ = exports.RuntimeHintValue$ = exports.RuntimeHints$ = exports.RuntimeHintDetails$ = exports.RecognizeUtteranceResponse$ = exports.RecognizeUtteranceRequest$ = exports.RecognizeTextResponse$ = exports.RecognizeTextRequest$ = exports.RecognizedBotMember$ = exports.PutSessionResponse$ = exports.PutSessionRequest$ = exports.PlaybackInterruptionEvent$ = exports.PlaybackCompletionEvent$ = exports.Message$ = exports.Interpretation$ = exports.IntentResultEvent$ = exports.Intent$ = exports.ImageResponseCard$ = exports.HeartbeatEvent$ = exports.GetSessionResponse$ = exports.GetSessionRequest$ = exports.ElicitSubSlot$ = exports.DTMFInputEvent$ = exports.DisconnectionEvent$ = exports.DialogAction$ = exports.DeleteSessionResponse$ = exports.DeleteSessionRequest$ = exports.ConfigurationEvent$ = exports.ConfidenceScore$ = exports.Button$ = exports.AudioResponseEvent$ = exports.AudioInputEvent$ = exports.ActiveContextTimeToLive$ = exports.ActiveContext$ = exports.errorTypeRegistries = exports.ValidationException$ = exports.ThrottlingException$ = exports.ResourceNotFoundException$ = exports.InternalServerException$ = exports.DependencyFailedException$ = exports.ConflictException$ = exports.BadGatewayException$ = exports.AccessDeniedException$ = exports.LexRuntimeV2ServiceException$ = void 0;
exports.StartConversation$ = exports.RecognizeUtterance$ = exports.RecognizeText$ = exports.PutSession$ = exports.GetSession$ = exports.DeleteSession$ = exports.StartConversationResponseEventStream$ = exports.StartConversationRequestEventStream$ = exports.Value$ = exports.TranscriptEvent$ = exports.TextResponseEvent$ = void 0;
const _AC = "ActiveContext";
const _ACL = "ActiveContextsList";
const _ACPM = "ActiveContextParametersMap";
const _ACTTL = "ActiveContextTimeToLive";
const _ADE = "AccessDeniedException";
const _AIE = "AudioInputEvent";
const _ARE = "AudioResponseEvent";
const _B = "Button";
const _BGE = "BadGatewayException";
const _BL = "ButtonsList";
const _BS = "BlobStream";
const _CE = "ConflictException";
const _CEo = "ConfigurationEvent";
const _CS = "ConfidenceScore";
const _CT = "Content-Type";
const _DA = "DialogAction";
const _DE = "DisconnectionEvent";
const _DFE = "DependencyFailedException";
const _DS = "DeleteSession";
const _DSR = "DeleteSessionRequest";
const _DSRe = "DeleteSessionResponse";
const _DTMFIE = "DTMFInputEvent";
const _DTMFR = "DTMFRegex";
const _ESS = "ElicitSubSlot";
const _GS = "GetSession";
const _GSR = "GetSessionRequest";
const _GSRe = "GetSessionResponse";
const _HE = "HeartbeatEvent";
const _I = "Intent";
const _IRC = "ImageResponseCard";
const _IRE = "IntentResultEvent";
const _ISE = "InternalServerException";
const _In = "Interpretation";
const _Int = "Interpretations";
const _M = "Message";
const _Me = "Messages";
const _PCE = "PlaybackCompletionEvent";
const _PIE = "PlaybackInterruptionEvent";
const _PS = "PutSession";
const _PSR = "PutSessionRequest";
const _PSRu = "PutSessionResponse";
const _RBM = "RecognizedBotMember";
const _RCT = "ResponseContentType";
const _RCT_ = "Response-Content-Type";
const _RH = "RuntimeHints";
const _RHD = "RuntimeHintDetails";
const _RHV = "RuntimeHintValue";
const _RHVL = "RuntimeHintValuesList";
const _RNFE = "ResourceNotFoundException";
const _RT = "RecognizeText";
const _RTR = "RecognizeTextRequest";
const _RTRe = "RecognizeTextResponse";
const _RU = "RecognizeUtterance";
const _RUR = "RecognizeUtteranceRequest";
const _RURe = "RecognizeUtteranceResponse";
const _S = "Slot";
const _SC = "StartConversation";
const _SCR = "StartConversationRequest";
const _SCRES = "StartConversationRequestEventStream";
const _SCRESt = "StartConversationResponseEventStream";
const _SCRt = "StartConversationResponse";
const _SHIM = "SlotHintsIntentMap";
const _SHSM = "SlotHintsSlotMap";
const _SNES = "SensitiveNonEmptyString";
const _SR = "SentimentResponse";
const _SS = "SentimentScore";
const _SSe = "SessionState";
const _Sl = "Slots";
const _T = "Text";
const _TE = "ThrottlingException";
const _TEr = "TranscriptEvent";
const _TIE = "TextInputEvent";
const _TRE = "TextResponseEvent";
const _V = "Value";
const _VE = "ValidationException";
const _Va = "Values";
const _aC = "audioChunk";
const _aCc = "activeContexts";
const _aS = "audioStream";
const _b = "buttons";
const _bAI = "botAliasId";
const _bI = "botId";
const _bN = "botName";
const _c = "client";
const _cA = "contextAttributes";
const _cBEI = "causedByEventId";
const _cM = "conversationMode";
const _cS = "confirmationState";
const _cT = "contentType";
const _cTM = "clientTimestampMillis";
const _co = "content";
const _dA = "dialogAction";
const _dP = "disablePlayback";
const _e = "error";
const _eI = "eventId";
const _eR = "eventReason";
const _h = "http";
const _hE = "httpError";
const _hH = "httpHeader";
const _i = "interpretations";
const _iC = "inputCharacter";
const _iM = "inputMode";
const _iRC = "imageResponseCard";
const _iS = "interpretationSource";
const _iSn = "inputStream";
const _iT = "inputTranscript";
const _iU = "imageUrl";
const _iV = "interpretedValue";
const _in = "intent";
const _lI = "localeId";
const _m = "message";
const _me = "messages";
const _mi = "mixed";
const _n = "name";
const _nC = "nluConfidence";
const _ne = "negative";
const _neu = "neutral";
const _oRI = "originatingRequestId";
const _oV = "originalValue";
const _p = "phrase";
const _po = "positive";
const _rA = "requestAttributes";
const _rBM = "recognizedBotMember";
const _rCT = "responseContentType";
const _rCTe = "requestContentType";
const _rES = "requestEventStream";
const _rESe = "responseEventStream";
const _rH = "runtimeHints";
const _rHV = "runtimeHintValues";
const _rV = "resolvedValues";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.lexruntimev2";
const _sA = "sessionAttributes";
const _sES = "slotElicitationStyle";
const _sH = "slotHints";
const _sI = "sessionId";
const _sR = "sentimentResponse";
const _sS = "sessionState";
const _sSH = "subSlotHints";
const _sSTE = "subSlotToElicit";
const _sSe = "sentimentScore";
const _sSu = "subSlots";
const _sTE = "slotToElicit";
const _sc = "score";
const _se = "server";
const _sen = "sentiment";
const _sh = "shape";
const _sl = "slots";
const _st = "streaming";
const _sta = "state";
const _su = "subtitle";
const _t = "text";
const _tTL = "timeToLive";
const _tTLIS = "timeToLiveInSeconds";
const _tTLu = "turnsToLive";
const _ti = "title";
const _tr = "transcript";
const _ty = "type";
const _v = "value";
const _va = "values";
const _wM = "welcomeMessages";
const _xalcm = "x-amz-lex-conversation-mode";
const _xali = "x-amz-lex-interpretations";
const _xalim = "x-amz-lex-input-mode";
const _xalit = "x-amz-lex-input-transcript";
const _xalm = "x-amz-lex-messages";
const _xalra = "x-amz-lex-request-attributes";
const _xalrbm = "x-amz-lex-recognized-bot-member";
const _xalsi = "x-amz-lex-session-id";
const _xalss = "x-amz-lex-session-state";
const n0 = "com.amazonaws.lexruntimev2";
const schema_1 = require("@smithy/core/schema");
const errors_1 = require("../models/errors");
const LexRuntimeV2ServiceException_1 = require("../models/LexRuntimeV2ServiceException");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.LexRuntimeV2ServiceException$ = [-3, _s, "LexRuntimeV2ServiceException", 0, [], []];
_s_registry.registerError(exports.LexRuntimeV2ServiceException$, LexRuntimeV2ServiceException_1.LexRuntimeV2ServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.AccessDeniedException$, errors_1.AccessDeniedException);
exports.BadGatewayException$ = [-3, n0, _BGE,
    { [_e]: _se, [_hE]: 502 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.BadGatewayException$, errors_1.BadGatewayException);
exports.ConflictException$ = [-3, n0, _CE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.ConflictException$, errors_1.ConflictException);
exports.DependencyFailedException$ = [-3, n0, _DFE,
    { [_e]: _c, [_hE]: 424 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.DependencyFailedException$, errors_1.DependencyFailedException);
exports.InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _se, [_hE]: 500 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.InternalServerException$, errors_1.InternalServerException);
exports.ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.ResourceNotFoundException$, errors_1.ResourceNotFoundException);
exports.ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c, [_hE]: 429 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.ThrottlingException$, errors_1.ThrottlingException);
exports.ValidationException$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0], 1
];
n0_registry.registerError(exports.ValidationException$, errors_1.ValidationException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var BlobStream = [0, n0, _BS, { [_st]: 1 }, 42];
var DTMFRegex = [0, n0, _DTMFR, 8, 0];
var SensitiveNonEmptyString = [0, n0, _SNES, 8, 0];
var Text = [0, n0, _T, 8, 0];
exports.ActiveContext$ = [3, n0, _AC,
    0,
    [_n, _tTL, _cA],
    [0, () => exports.ActiveContextTimeToLive$, [() => ActiveContextParametersMap, 0]], 3
];
exports.ActiveContextTimeToLive$ = [3, n0, _ACTTL,
    0,
    [_tTLIS, _tTLu],
    [1, 1], 2
];
exports.AudioInputEvent$ = [3, n0, _AIE,
    0,
    [_cT, _aC, _eI, _cTM],
    [0, 21, 0, 1], 1
];
exports.AudioResponseEvent$ = [3, n0, _ARE,
    0,
    [_aC, _cT, _eI],
    [21, 0, 0]
];
exports.Button$ = [3, n0, _B,
    0,
    [_t, _v],
    [0, 0], 2
];
exports.ConfidenceScore$ = [3, n0, _CS,
    0,
    [_sc],
    [1]
];
exports.ConfigurationEvent$ = [3, n0, _CEo,
    0,
    [_rCT, _rA, _sS, _wM, _dP, _eI, _cTM],
    [0, 128 | 0, [() => exports.SessionState$, 0], [() => Messages, 0], 2, 0, 1], 1
];
exports.DeleteSessionRequest$ = [3, n0, _DSR,
    0,
    [_bI, _bAI, _lI, _sI],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
exports.DeleteSessionResponse$ = [3, n0, _DSRe,
    0,
    [_bI, _bAI, _lI, _sI],
    [0, 0, 0, 0]
];
exports.DialogAction$ = [3, n0, _DA,
    0,
    [_ty, _sTE, _sES, _sSTE],
    [0, 0, 0, () => exports.ElicitSubSlot$], 1
];
exports.DisconnectionEvent$ = [3, n0, _DE,
    0,
    [_eI, _cTM],
    [0, 1]
];
exports.DTMFInputEvent$ = [3, n0, _DTMFIE,
    0,
    [_iC, _eI, _cTM],
    [[() => DTMFRegex, 0], 0, 1], 1
];
exports.ElicitSubSlot$ = [3, n0, _ESS,
    0,
    [_n, _sSTE],
    [0, () => exports.ElicitSubSlot$], 1
];
exports.GetSessionRequest$ = [3, n0, _GSR,
    0,
    [_bI, _bAI, _lI, _sI],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
exports.GetSessionResponse$ = [3, n0, _GSRe,
    0,
    [_sI, _me, _i, _sS],
    [0, [() => Messages, 0], () => Interpretations, [() => exports.SessionState$, 0]]
];
exports.HeartbeatEvent$ = [3, n0, _HE,
    0,
    [_eI],
    [0]
];
exports.ImageResponseCard$ = [3, n0, _IRC,
    0,
    [_ti, _su, _iU, _b],
    [0, 0, 0, () => ButtonsList], 1
];
exports.Intent$ = [3, n0, _I,
    0,
    [_n, _sl, _sta, _cS],
    [0, () => Slots, 0, 0], 1
];
exports.IntentResultEvent$ = [3, n0, _IRE,
    0,
    [_iM, _i, _sS, _rA, _sI, _eI, _rBM],
    [0, () => Interpretations, [() => exports.SessionState$, 0], 128 | 0, 0, 0, () => exports.RecognizedBotMember$]
];
exports.Interpretation$ = [3, n0, _In,
    0,
    [_nC, _sR, _in, _iS],
    [() => exports.ConfidenceScore$, () => exports.SentimentResponse$, () => exports.Intent$, 0]
];
exports.Message$ = [3, n0, _M,
    0,
    [_cT, _co, _iRC],
    [0, [() => Text, 0], () => exports.ImageResponseCard$], 1
];
exports.PlaybackCompletionEvent$ = [3, n0, _PCE,
    0,
    [_eI, _cTM],
    [0, 1]
];
exports.PlaybackInterruptionEvent$ = [3, n0, _PIE,
    0,
    [_eR, _cBEI, _eI],
    [0, 0, 0]
];
exports.PutSessionRequest$ = [3, n0, _PSR,
    0,
    [_bI, _bAI, _lI, _sI, _sS, _me, _rA, _rCT],
    [[0, 1], [0, 1], [0, 1], [0, 1], [() => exports.SessionState$, 0], [() => Messages, 0], 128 | 0, [0, { [_hH]: _RCT }]], 5
];
exports.PutSessionResponse$ = [3, n0, _PSRu,
    0,
    [_cT, _me, _sS, _rA, _sI, _aS],
    [[0, { [_hH]: _CT }], [0, { [_hH]: _xalm }], [0, { [_hH]: _xalss }], [0, { [_hH]: _xalra }], [0, { [_hH]: _xalsi }], [() => BlobStream, 16]]
];
exports.RecognizedBotMember$ = [3, n0, _RBM,
    0,
    [_bI, _bN],
    [0, 0], 1
];
exports.RecognizeTextRequest$ = [3, n0, _RTR,
    0,
    [_bI, _bAI, _lI, _sI, _t, _sS, _rA],
    [[0, 1], [0, 1], [0, 1], [0, 1], [() => Text, 0], [() => exports.SessionState$, 0], 128 | 0], 5
];
exports.RecognizeTextResponse$ = [3, n0, _RTRe,
    0,
    [_me, _sS, _i, _rA, _sI, _rBM],
    [[() => Messages, 0], [() => exports.SessionState$, 0], () => Interpretations, 128 | 0, 0, () => exports.RecognizedBotMember$]
];
exports.RecognizeUtteranceRequest$ = [3, n0, _RUR,
    0,
    [_bI, _bAI, _lI, _sI, _rCTe, _sS, _rA, _rCT, _iSn],
    [[0, 1], [0, 1], [0, 1], [0, 1], [0, { [_hH]: _CT }], [() => SensitiveNonEmptyString, { [_hH]: _xalss }], [() => SensitiveNonEmptyString, { [_hH]: _xalra }], [0, { [_hH]: _RCT_ }], [() => BlobStream, 16]], 5
];
exports.RecognizeUtteranceResponse$ = [3, n0, _RURe,
    0,
    [_iM, _cT, _me, _i, _sS, _rA, _sI, _iT, _aS, _rBM],
    [[0, { [_hH]: _xalim }], [0, { [_hH]: _CT }], [0, { [_hH]: _xalm }], [0, { [_hH]: _xali }], [0, { [_hH]: _xalss }], [0, { [_hH]: _xalra }], [0, { [_hH]: _xalsi }], [0, { [_hH]: _xalit }], [() => BlobStream, 16], [0, { [_hH]: _xalrbm }]]
];
exports.RuntimeHintDetails$ = [3, n0, _RHD,
    0,
    [_rHV, _sSH],
    [() => RuntimeHintValuesList, () => SlotHintsSlotMap]
];
exports.RuntimeHints$ = [3, n0, _RH,
    0,
    [_sH],
    [() => SlotHintsIntentMap]
];
exports.RuntimeHintValue$ = [3, n0, _RHV,
    0,
    [_p],
    [0], 1
];
exports.SentimentResponse$ = [3, n0, _SR,
    0,
    [_sen, _sSe],
    [0, () => exports.SentimentScore$]
];
exports.SentimentScore$ = [3, n0, _SS,
    0,
    [_po, _ne, _neu, _mi],
    [1, 1, 1, 1]
];
exports.SessionState$ = [3, n0, _SSe,
    0,
    [_dA, _in, _aCc, _sA, _oRI, _rH],
    [() => exports.DialogAction$, () => exports.Intent$, [() => ActiveContextsList, 0], 128 | 0, 0, () => exports.RuntimeHints$]
];
exports.Slot$ = [3, n0, _S,
    0,
    [_v, _sh, _va, _sSu],
    [() => exports.Value$, 0, () => Values, () => Slots]
];
exports.StartConversationRequest$ = [3, n0, _SCR,
    0,
    [_bI, _bAI, _lI, _sI, _rES, _cM],
    [[0, 1], [0, 1], [0, 1], [0, 1], [() => exports.StartConversationRequestEventStream$, 16], [0, { [_hH]: _xalcm }]], 5
];
exports.StartConversationResponse$ = [3, n0, _SCRt,
    0,
    [_rESe],
    [[() => exports.StartConversationResponseEventStream$, 16]]
];
exports.TextInputEvent$ = [3, n0, _TIE,
    0,
    [_t, _eI, _cTM],
    [[() => Text, 0], 0, 1], 1
];
exports.TextResponseEvent$ = [3, n0, _TRE,
    0,
    [_me, _eI],
    [[() => Messages, 0], 0]
];
exports.TranscriptEvent$ = [3, n0, _TEr,
    0,
    [_tr, _eI],
    [0, 0]
];
exports.Value$ = [3, n0, _V,
    0,
    [_iV, _oV, _rV],
    [0, 0, 64 | 0], 1
];
var ActiveContextsList = [1, n0, _ACL,
    0, [() => exports.ActiveContext$,
        0]
];
var ButtonsList = [1, n0, _BL,
    0, () => exports.Button$
];
var Interpretations = [1, n0, _Int,
    0, () => exports.Interpretation$
];
var Messages = [1, n0, _Me,
    0, [() => exports.Message$,
        0]
];
var RuntimeHintValuesList = [1, n0, _RHVL,
    0, () => exports.RuntimeHintValue$
];
var StringList = 64 | 0;
var Values = [1, n0, _Va,
    0, () => exports.Slot$
];
var ActiveContextParametersMap = [2, n0, _ACPM,
    0, [0,
        0],
    [() => Text,
        0]
];
var SlotHintsIntentMap = [2, n0, _SHIM,
    0, 0, () => SlotHintsSlotMap
];
var SlotHintsSlotMap = [2, n0, _SHSM,
    0, 0, () => exports.RuntimeHintDetails$
];
var Slots = [2, n0, _Sl,
    0, 0, () => exports.Slot$
];
var StringMap = 128 | 0;
exports.StartConversationRequestEventStream$ = [4, n0, _SCRES,
    { [_st]: 1 },
    [_CEo, _AIE, _DTMFIE, _TIE, _PCE, _DE],
    [[() => exports.ConfigurationEvent$, 0], () => exports.AudioInputEvent$, [() => exports.DTMFInputEvent$, 0], [() => exports.TextInputEvent$, 0], () => exports.PlaybackCompletionEvent$, () => exports.DisconnectionEvent$]
];
exports.StartConversationResponseEventStream$ = [4, n0, _SCRESt,
    { [_st]: 1 },
    [_PIE, _TEr, _IRE, _TRE, _ARE, _HE, _ADE, _RNFE, _VE, _TE, _ISE, _CE, _DFE, _BGE],
    [() => exports.PlaybackInterruptionEvent$, () => exports.TranscriptEvent$, [() => exports.IntentResultEvent$, 0], [() => exports.TextResponseEvent$, 0], () => exports.AudioResponseEvent$, () => exports.HeartbeatEvent$, [() => exports.AccessDeniedException$, 0], [() => exports.ResourceNotFoundException$, 0], [() => exports.ValidationException$, 0], [() => exports.ThrottlingException$, 0], [() => exports.InternalServerException$, 0], [() => exports.ConflictException$, 0], [() => exports.DependencyFailedException$, 0], [() => exports.BadGatewayException$, 0]]
];
exports.DeleteSession$ = [9, n0, _DS,
    { [_h]: ["DELETE", "/bots/{botId}/botAliases/{botAliasId}/botLocales/{localeId}/sessions/{sessionId}", 200] }, () => exports.DeleteSessionRequest$, () => exports.DeleteSessionResponse$
];
exports.GetSession$ = [9, n0, _GS,
    { [_h]: ["GET", "/bots/{botId}/botAliases/{botAliasId}/botLocales/{localeId}/sessions/{sessionId}", 200] }, () => exports.GetSessionRequest$, () => exports.GetSessionResponse$
];
exports.PutSession$ = [9, n0, _PS,
    { [_h]: ["POST", "/bots/{botId}/botAliases/{botAliasId}/botLocales/{localeId}/sessions/{sessionId}", 200] }, () => exports.PutSessionRequest$, () => exports.PutSessionResponse$
];
exports.RecognizeText$ = [9, n0, _RT,
    { [_h]: ["POST", "/bots/{botId}/botAliases/{botAliasId}/botLocales/{localeId}/sessions/{sessionId}/text", 200] }, () => exports.RecognizeTextRequest$, () => exports.RecognizeTextResponse$
];
exports.RecognizeUtterance$ = [9, n0, _RU,
    { [_h]: ["POST", "/bots/{botId}/botAliases/{botAliasId}/botLocales/{localeId}/sessions/{sessionId}/utterance", 200] }, () => exports.RecognizeUtteranceRequest$, () => exports.RecognizeUtteranceResponse$
];
exports.StartConversation$ = [9, n0, _SC,
    { [_h]: ["POST", "/bots/{botId}/botAliases/{botAliasId}/botLocales/{localeId}/sessions/{sessionId}/conversation", 200] }, () => exports.StartConversationRequest$, () => exports.StartConversationResponse$
];
