// @aws-sdk/client-location/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var LocationServiceException = require('./models/LocationServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "geo",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class LocationClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultLocationHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AssociateTrackerConsumerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "AssociateTrackerConsumer", {})
    .n("LocationClient", "AssociateTrackerConsumerCommand")
    .sc(schemas_0.AssociateTrackerConsumer$)
    .build() {
}

class BatchDeleteDevicePositionHistoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "BatchDeleteDevicePositionHistory", {})
    .n("LocationClient", "BatchDeleteDevicePositionHistoryCommand")
    .sc(schemas_0.BatchDeleteDevicePositionHistory$)
    .build() {
}

class BatchDeleteGeofenceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "BatchDeleteGeofence", {})
    .n("LocationClient", "BatchDeleteGeofenceCommand")
    .sc(schemas_0.BatchDeleteGeofence$)
    .build() {
}

class BatchEvaluateGeofencesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "BatchEvaluateGeofences", {})
    .n("LocationClient", "BatchEvaluateGeofencesCommand")
    .sc(schemas_0.BatchEvaluateGeofences$)
    .build() {
}

class BatchGetDevicePositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "BatchGetDevicePosition", {})
    .n("LocationClient", "BatchGetDevicePositionCommand")
    .sc(schemas_0.BatchGetDevicePosition$)
    .build() {
}

class BatchPutGeofenceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "BatchPutGeofence", {})
    .n("LocationClient", "BatchPutGeofenceCommand")
    .sc(schemas_0.BatchPutGeofence$)
    .build() {
}

class BatchUpdateDevicePositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "BatchUpdateDevicePosition", {})
    .n("LocationClient", "BatchUpdateDevicePositionCommand")
    .sc(schemas_0.BatchUpdateDevicePosition$)
    .build() {
}

class CalculateRouteCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CalculateRoute", {})
    .n("LocationClient", "CalculateRouteCommand")
    .sc(schemas_0.CalculateRoute$)
    .build() {
}

class CalculateRouteMatrixCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CalculateRouteMatrix", {})
    .n("LocationClient", "CalculateRouteMatrixCommand")
    .sc(schemas_0.CalculateRouteMatrix$)
    .build() {
}

class CreateGeofenceCollectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CreateGeofenceCollection", {})
    .n("LocationClient", "CreateGeofenceCollectionCommand")
    .sc(schemas_0.CreateGeofenceCollection$)
    .build() {
}

class CreateKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CreateKey", {})
    .n("LocationClient", "CreateKeyCommand")
    .sc(schemas_0.CreateKey$)
    .build() {
}

class CreateMapCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CreateMap", {})
    .n("LocationClient", "CreateMapCommand")
    .sc(schemas_0.CreateMap$)
    .build() {
}

class CreatePlaceIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CreatePlaceIndex", {})
    .n("LocationClient", "CreatePlaceIndexCommand")
    .sc(schemas_0.CreatePlaceIndex$)
    .build() {
}

class CreateRouteCalculatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CreateRouteCalculator", {})
    .n("LocationClient", "CreateRouteCalculatorCommand")
    .sc(schemas_0.CreateRouteCalculator$)
    .build() {
}

class CreateTrackerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "CreateTracker", {})
    .n("LocationClient", "CreateTrackerCommand")
    .sc(schemas_0.CreateTracker$)
    .build() {
}

class DeleteGeofenceCollectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DeleteGeofenceCollection", {})
    .n("LocationClient", "DeleteGeofenceCollectionCommand")
    .sc(schemas_0.DeleteGeofenceCollection$)
    .build() {
}

class DeleteKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DeleteKey", {})
    .n("LocationClient", "DeleteKeyCommand")
    .sc(schemas_0.DeleteKey$)
    .build() {
}

class DeleteMapCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DeleteMap", {})
    .n("LocationClient", "DeleteMapCommand")
    .sc(schemas_0.DeleteMap$)
    .build() {
}

class DeletePlaceIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DeletePlaceIndex", {})
    .n("LocationClient", "DeletePlaceIndexCommand")
    .sc(schemas_0.DeletePlaceIndex$)
    .build() {
}

class DeleteRouteCalculatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DeleteRouteCalculator", {})
    .n("LocationClient", "DeleteRouteCalculatorCommand")
    .sc(schemas_0.DeleteRouteCalculator$)
    .build() {
}

class DeleteTrackerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DeleteTracker", {})
    .n("LocationClient", "DeleteTrackerCommand")
    .sc(schemas_0.DeleteTracker$)
    .build() {
}

class DescribeGeofenceCollectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DescribeGeofenceCollection", {})
    .n("LocationClient", "DescribeGeofenceCollectionCommand")
    .sc(schemas_0.DescribeGeofenceCollection$)
    .build() {
}

class DescribeKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DescribeKey", {})
    .n("LocationClient", "DescribeKeyCommand")
    .sc(schemas_0.DescribeKey$)
    .build() {
}

class DescribeMapCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DescribeMap", {})
    .n("LocationClient", "DescribeMapCommand")
    .sc(schemas_0.DescribeMap$)
    .build() {
}

class DescribePlaceIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DescribePlaceIndex", {})
    .n("LocationClient", "DescribePlaceIndexCommand")
    .sc(schemas_0.DescribePlaceIndex$)
    .build() {
}

class DescribeRouteCalculatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DescribeRouteCalculator", {})
    .n("LocationClient", "DescribeRouteCalculatorCommand")
    .sc(schemas_0.DescribeRouteCalculator$)
    .build() {
}

class DescribeTrackerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DescribeTracker", {})
    .n("LocationClient", "DescribeTrackerCommand")
    .sc(schemas_0.DescribeTracker$)
    .build() {
}

class DisassociateTrackerConsumerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "DisassociateTrackerConsumer", {})
    .n("LocationClient", "DisassociateTrackerConsumerCommand")
    .sc(schemas_0.DisassociateTrackerConsumer$)
    .build() {
}

class ForecastGeofenceEventsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ForecastGeofenceEvents", {})
    .n("LocationClient", "ForecastGeofenceEventsCommand")
    .sc(schemas_0.ForecastGeofenceEvents$)
    .build() {
}

class GetDevicePositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetDevicePosition", {})
    .n("LocationClient", "GetDevicePositionCommand")
    .sc(schemas_0.GetDevicePosition$)
    .build() {
}

class GetDevicePositionHistoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetDevicePositionHistory", {})
    .n("LocationClient", "GetDevicePositionHistoryCommand")
    .sc(schemas_0.GetDevicePositionHistory$)
    .build() {
}

class GetGeofenceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetGeofence", {})
    .n("LocationClient", "GetGeofenceCommand")
    .sc(schemas_0.GetGeofence$)
    .build() {
}

class GetMapGlyphsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetMapGlyphs", {})
    .n("LocationClient", "GetMapGlyphsCommand")
    .sc(schemas_0.GetMapGlyphs$)
    .build() {
}

class GetMapSpritesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetMapSprites", {})
    .n("LocationClient", "GetMapSpritesCommand")
    .sc(schemas_0.GetMapSprites$)
    .build() {
}

class GetMapStyleDescriptorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetMapStyleDescriptor", {})
    .n("LocationClient", "GetMapStyleDescriptorCommand")
    .sc(schemas_0.GetMapStyleDescriptor$)
    .build() {
}

class GetMapTileCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetMapTile", {})
    .n("LocationClient", "GetMapTileCommand")
    .sc(schemas_0.GetMapTile$)
    .build() {
}

class GetPlaceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "GetPlace", {})
    .n("LocationClient", "GetPlaceCommand")
    .sc(schemas_0.GetPlace$)
    .build() {
}

class ListDevicePositionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListDevicePositions", {})
    .n("LocationClient", "ListDevicePositionsCommand")
    .sc(schemas_0.ListDevicePositions$)
    .build() {
}

class ListGeofenceCollectionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListGeofenceCollections", {})
    .n("LocationClient", "ListGeofenceCollectionsCommand")
    .sc(schemas_0.ListGeofenceCollections$)
    .build() {
}

class ListGeofencesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListGeofences", {})
    .n("LocationClient", "ListGeofencesCommand")
    .sc(schemas_0.ListGeofences$)
    .build() {
}

class ListKeysCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListKeys", {})
    .n("LocationClient", "ListKeysCommand")
    .sc(schemas_0.ListKeys$)
    .build() {
}

class ListMapsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListMaps", {})
    .n("LocationClient", "ListMapsCommand")
    .sc(schemas_0.ListMaps$)
    .build() {
}

class ListPlaceIndexesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListPlaceIndexes", {})
    .n("LocationClient", "ListPlaceIndexesCommand")
    .sc(schemas_0.ListPlaceIndexes$)
    .build() {
}

class ListRouteCalculatorsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListRouteCalculators", {})
    .n("LocationClient", "ListRouteCalculatorsCommand")
    .sc(schemas_0.ListRouteCalculators$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListTagsForResource", {})
    .n("LocationClient", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListTrackerConsumersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListTrackerConsumers", {})
    .n("LocationClient", "ListTrackerConsumersCommand")
    .sc(schemas_0.ListTrackerConsumers$)
    .build() {
}

class ListTrackersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "ListTrackers", {})
    .n("LocationClient", "ListTrackersCommand")
    .sc(schemas_0.ListTrackers$)
    .build() {
}

class PutGeofenceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "PutGeofence", {})
    .n("LocationClient", "PutGeofenceCommand")
    .sc(schemas_0.PutGeofence$)
    .build() {
}

class SearchPlaceIndexForPositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "SearchPlaceIndexForPosition", {})
    .n("LocationClient", "SearchPlaceIndexForPositionCommand")
    .sc(schemas_0.SearchPlaceIndexForPosition$)
    .build() {
}

class SearchPlaceIndexForSuggestionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "SearchPlaceIndexForSuggestions", {})
    .n("LocationClient", "SearchPlaceIndexForSuggestionsCommand")
    .sc(schemas_0.SearchPlaceIndexForSuggestions$)
    .build() {
}

class SearchPlaceIndexForTextCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "SearchPlaceIndexForText", {})
    .n("LocationClient", "SearchPlaceIndexForTextCommand")
    .sc(schemas_0.SearchPlaceIndexForText$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "TagResource", {})
    .n("LocationClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "UntagResource", {})
    .n("LocationClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateGeofenceCollectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "UpdateGeofenceCollection", {})
    .n("LocationClient", "UpdateGeofenceCollectionCommand")
    .sc(schemas_0.UpdateGeofenceCollection$)
    .build() {
}

class UpdateKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "UpdateKey", {})
    .n("LocationClient", "UpdateKeyCommand")
    .sc(schemas_0.UpdateKey$)
    .build() {
}

class UpdateMapCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "UpdateMap", {})
    .n("LocationClient", "UpdateMapCommand")
    .sc(schemas_0.UpdateMap$)
    .build() {
}

class UpdatePlaceIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "UpdatePlaceIndex", {})
    .n("LocationClient", "UpdatePlaceIndexCommand")
    .sc(schemas_0.UpdatePlaceIndex$)
    .build() {
}

class UpdateRouteCalculatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "UpdateRouteCalculator", {})
    .n("LocationClient", "UpdateRouteCalculatorCommand")
    .sc(schemas_0.UpdateRouteCalculator$)
    .build() {
}

class UpdateTrackerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "UpdateTracker", {})
    .n("LocationClient", "UpdateTrackerCommand")
    .sc(schemas_0.UpdateTracker$)
    .build() {
}

class VerifyDevicePositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("LocationService", "VerifyDevicePosition", {})
    .n("LocationClient", "VerifyDevicePositionCommand")
    .sc(schemas_0.VerifyDevicePosition$)
    .build() {
}

const paginateForecastGeofenceEvents = core.createPaginator(LocationClient, ForecastGeofenceEventsCommand, "NextToken", "NextToken", "MaxResults");

const paginateGetDevicePositionHistory = core.createPaginator(LocationClient, GetDevicePositionHistoryCommand, "NextToken", "NextToken", "MaxResults");

const paginateListDevicePositions = core.createPaginator(LocationClient, ListDevicePositionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListGeofenceCollections = core.createPaginator(LocationClient, ListGeofenceCollectionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListGeofences = core.createPaginator(LocationClient, ListGeofencesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListKeys = core.createPaginator(LocationClient, ListKeysCommand, "NextToken", "NextToken", "MaxResults");

const paginateListMaps = core.createPaginator(LocationClient, ListMapsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListPlaceIndexes = core.createPaginator(LocationClient, ListPlaceIndexesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListRouteCalculators = core.createPaginator(LocationClient, ListRouteCalculatorsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTrackerConsumers = core.createPaginator(LocationClient, ListTrackerConsumersCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTrackers = core.createPaginator(LocationClient, ListTrackersCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    AssociateTrackerConsumerCommand,
    BatchDeleteDevicePositionHistoryCommand,
    BatchDeleteGeofenceCommand,
    BatchEvaluateGeofencesCommand,
    BatchGetDevicePositionCommand,
    BatchPutGeofenceCommand,
    BatchUpdateDevicePositionCommand,
    CalculateRouteCommand,
    CalculateRouteMatrixCommand,
    CreateGeofenceCollectionCommand,
    CreateKeyCommand,
    CreateMapCommand,
    CreatePlaceIndexCommand,
    CreateRouteCalculatorCommand,
    CreateTrackerCommand,
    DeleteGeofenceCollectionCommand,
    DeleteKeyCommand,
    DeleteMapCommand,
    DeletePlaceIndexCommand,
    DeleteRouteCalculatorCommand,
    DeleteTrackerCommand,
    DescribeGeofenceCollectionCommand,
    DescribeKeyCommand,
    DescribeMapCommand,
    DescribePlaceIndexCommand,
    DescribeRouteCalculatorCommand,
    DescribeTrackerCommand,
    DisassociateTrackerConsumerCommand,
    ForecastGeofenceEventsCommand,
    GetDevicePositionCommand,
    GetDevicePositionHistoryCommand,
    GetGeofenceCommand,
    GetMapGlyphsCommand,
    GetMapSpritesCommand,
    GetMapStyleDescriptorCommand,
    GetMapTileCommand,
    GetPlaceCommand,
    ListDevicePositionsCommand,
    ListGeofenceCollectionsCommand,
    ListGeofencesCommand,
    ListKeysCommand,
    ListMapsCommand,
    ListPlaceIndexesCommand,
    ListRouteCalculatorsCommand,
    ListTagsForResourceCommand,
    ListTrackerConsumersCommand,
    ListTrackersCommand,
    PutGeofenceCommand,
    SearchPlaceIndexForPositionCommand,
    SearchPlaceIndexForSuggestionsCommand,
    SearchPlaceIndexForTextCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateGeofenceCollectionCommand,
    UpdateKeyCommand,
    UpdateMapCommand,
    UpdatePlaceIndexCommand,
    UpdateRouteCalculatorCommand,
    UpdateTrackerCommand,
    VerifyDevicePositionCommand,
};
const paginators = {
    paginateForecastGeofenceEvents,
    paginateGetDevicePositionHistory,
    paginateListDevicePositions,
    paginateListGeofenceCollections,
    paginateListGeofences,
    paginateListKeys,
    paginateListMaps,
    paginateListPlaceIndexes,
    paginateListRouteCalculators,
    paginateListTrackerConsumers,
    paginateListTrackers,
};
class Location extends LocationClient {
}
smithyClient.createAggregatedClient(commands, Location, { paginators });

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
Object.defineProperty(exports, "LocationServiceException", {
    enumerable: true,
    get: function () { return LocationServiceException.LocationServiceException; }
});
exports.AssociateTrackerConsumerCommand = AssociateTrackerConsumerCommand;
exports.BatchDeleteDevicePositionHistoryCommand = BatchDeleteDevicePositionHistoryCommand;
exports.BatchDeleteGeofenceCommand = BatchDeleteGeofenceCommand;
exports.BatchEvaluateGeofencesCommand = BatchEvaluateGeofencesCommand;
exports.BatchGetDevicePositionCommand = BatchGetDevicePositionCommand;
exports.BatchPutGeofenceCommand = BatchPutGeofenceCommand;
exports.BatchUpdateDevicePositionCommand = BatchUpdateDevicePositionCommand;
exports.CalculateRouteCommand = CalculateRouteCommand;
exports.CalculateRouteMatrixCommand = CalculateRouteMatrixCommand;
exports.CreateGeofenceCollectionCommand = CreateGeofenceCollectionCommand;
exports.CreateKeyCommand = CreateKeyCommand;
exports.CreateMapCommand = CreateMapCommand;
exports.CreatePlaceIndexCommand = CreatePlaceIndexCommand;
exports.CreateRouteCalculatorCommand = CreateRouteCalculatorCommand;
exports.CreateTrackerCommand = CreateTrackerCommand;
exports.DeleteGeofenceCollectionCommand = DeleteGeofenceCollectionCommand;
exports.DeleteKeyCommand = DeleteKeyCommand;
exports.DeleteMapCommand = DeleteMapCommand;
exports.DeletePlaceIndexCommand = DeletePlaceIndexCommand;
exports.DeleteRouteCalculatorCommand = DeleteRouteCalculatorCommand;
exports.DeleteTrackerCommand = DeleteTrackerCommand;
exports.DescribeGeofenceCollectionCommand = DescribeGeofenceCollectionCommand;
exports.DescribeKeyCommand = DescribeKeyCommand;
exports.DescribeMapCommand = DescribeMapCommand;
exports.DescribePlaceIndexCommand = DescribePlaceIndexCommand;
exports.DescribeRouteCalculatorCommand = DescribeRouteCalculatorCommand;
exports.DescribeTrackerCommand = DescribeTrackerCommand;
exports.DisassociateTrackerConsumerCommand = DisassociateTrackerConsumerCommand;
exports.ForecastGeofenceEventsCommand = ForecastGeofenceEventsCommand;
exports.GetDevicePositionCommand = GetDevicePositionCommand;
exports.GetDevicePositionHistoryCommand = GetDevicePositionHistoryCommand;
exports.GetGeofenceCommand = GetGeofenceCommand;
exports.GetMapGlyphsCommand = GetMapGlyphsCommand;
exports.GetMapSpritesCommand = GetMapSpritesCommand;
exports.GetMapStyleDescriptorCommand = GetMapStyleDescriptorCommand;
exports.GetMapTileCommand = GetMapTileCommand;
exports.GetPlaceCommand = GetPlaceCommand;
exports.ListDevicePositionsCommand = ListDevicePositionsCommand;
exports.ListGeofenceCollectionsCommand = ListGeofenceCollectionsCommand;
exports.ListGeofencesCommand = ListGeofencesCommand;
exports.ListKeysCommand = ListKeysCommand;
exports.ListMapsCommand = ListMapsCommand;
exports.ListPlaceIndexesCommand = ListPlaceIndexesCommand;
exports.ListRouteCalculatorsCommand = ListRouteCalculatorsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTrackerConsumersCommand = ListTrackerConsumersCommand;
exports.ListTrackersCommand = ListTrackersCommand;
exports.Location = Location;
exports.LocationClient = LocationClient;
exports.PutGeofenceCommand = PutGeofenceCommand;
exports.SearchPlaceIndexForPositionCommand = SearchPlaceIndexForPositionCommand;
exports.SearchPlaceIndexForSuggestionsCommand = SearchPlaceIndexForSuggestionsCommand;
exports.SearchPlaceIndexForTextCommand = SearchPlaceIndexForTextCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateGeofenceCollectionCommand = UpdateGeofenceCollectionCommand;
exports.UpdateKeyCommand = UpdateKeyCommand;
exports.UpdateMapCommand = UpdateMapCommand;
exports.UpdatePlaceIndexCommand = UpdatePlaceIndexCommand;
exports.UpdateRouteCalculatorCommand = UpdateRouteCalculatorCommand;
exports.UpdateTrackerCommand = UpdateTrackerCommand;
exports.VerifyDevicePositionCommand = VerifyDevicePositionCommand;
exports.paginateForecastGeofenceEvents = paginateForecastGeofenceEvents;
exports.paginateGetDevicePositionHistory = paginateGetDevicePositionHistory;
exports.paginateListDevicePositions = paginateListDevicePositions;
exports.paginateListGeofenceCollections = paginateListGeofenceCollections;
exports.paginateListGeofences = paginateListGeofences;
exports.paginateListKeys = paginateListKeys;
exports.paginateListMaps = paginateListMaps;
exports.paginateListPlaceIndexes = paginateListPlaceIndexes;
exports.paginateListRouteCalculators = paginateListRouteCalculators;
exports.paginateListTrackerConsumers = paginateListTrackerConsumers;
exports.paginateListTrackers = paginateListTrackers;
Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) Object.defineProperty(exports, k, {
        enumerable: true,
        get: function () { return schemas_0[k]; }
    });
});
Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) Object.defineProperty(exports, k, {
        enumerable: true,
        get: function () { return errors[k]; }
    });
});
