// @aws-sdk/client-lex-runtime-v2/dist-cjs/index.js
'use strict';

var middlewareEventstream = require('@aws-sdk/middleware-eventstream');
var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var eventstreamSerdeConfigResolver = require('@smithy/eventstream-serde-config-resolver');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var LexRuntimeV2ServiceException = require('./models/LexRuntimeV2ServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "lex",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class LexRuntimeV2Client extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = eventstreamSerdeConfigResolver.resolveEventStreamSerdeConfig(_config_6);
        const _config_8 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = middlewareEventstream.resolveEventStreamConfig(_config_8);
        const _config_10 = resolveRuntimeExtensions(_config_9, configuration?.extensions || []);
        this.config = _config_10;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultLexRuntimeV2HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class DeleteSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDeepSenseRunTimeServiceApi2_0", "DeleteSession", {})
    .n("LexRuntimeV2Client", "DeleteSessionCommand")
    .sc(schemas_0.DeleteSession$)
    .build() {
}

class GetSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDeepSenseRunTimeServiceApi2_0", "GetSession", {})
    .n("LexRuntimeV2Client", "GetSessionCommand")
    .sc(schemas_0.GetSession$)
    .build() {
}

class PutSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDeepSenseRunTimeServiceApi2_0", "PutSession", {})
    .n("LexRuntimeV2Client", "PutSessionCommand")
    .sc(schemas_0.PutSession$)
    .build() {
}

class RecognizeTextCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDeepSenseRunTimeServiceApi2_0", "RecognizeText", {})
    .n("LexRuntimeV2Client", "RecognizeTextCommand")
    .sc(schemas_0.RecognizeText$)
    .build() {
}

class RecognizeUtteranceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSDeepSenseRunTimeServiceApi2_0", "RecognizeUtterance", {})
    .n("LexRuntimeV2Client", "RecognizeUtteranceCommand")
    .sc(schemas_0.RecognizeUtterance$)
    .build() {
}

class StartConversationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        middlewareEventstream.getEventStreamPlugin(config),
    ];
})
    .s("AWSDeepSenseRunTimeServiceApi2_0", "StartConversation", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("LexRuntimeV2Client", "StartConversationCommand")
    .sc(schemas_0.StartConversation$)
    .build() {
}

const commands = {
    DeleteSessionCommand,
    GetSessionCommand,
    PutSessionCommand,
    RecognizeTextCommand,
    RecognizeUtteranceCommand,
    StartConversationCommand,
};
class LexRuntimeV2 extends LexRuntimeV2Client {
}
smithyClient.createAggregatedClient(commands, LexRuntimeV2);

const ConfirmationState = {
    CONFIRMED: "Confirmed",
    DENIED: "Denied",
    NONE: "None",
};
const Shape = {
    COMPOSITE: "Composite",
    LIST: "List",
    SCALAR: "Scalar",
};
const IntentState = {
    FAILED: "Failed",
    FULFILLED: "Fulfilled",
    FULFILLMENT_IN_PROGRESS: "FulfillmentInProgress",
    IN_PROGRESS: "InProgress",
    READY_FOR_FULFILLMENT: "ReadyForFulfillment",
    WAITING: "Waiting",
};
const InterpretationSource = {
    BEDROCK: "Bedrock",
    LEX: "Lex",
};
const SentimentType = {
    MIXED: "MIXED",
    NEGATIVE: "NEGATIVE",
    NEUTRAL: "NEUTRAL",
    POSITIVE: "POSITIVE",
};
const MessageContentType = {
    CUSTOM_PAYLOAD: "CustomPayload",
    IMAGE_RESPONSE_CARD: "ImageResponseCard",
    PLAIN_TEXT: "PlainText",
    SSML: "SSML",
};
const StyleType = {
    DEFAULT: "Default",
    SPELL_BY_LETTER: "SpellByLetter",
    SPELL_BY_WORD: "SpellByWord",
};
const DialogActionType = {
    CLOSE: "Close",
    CONFIRM_INTENT: "ConfirmIntent",
    DELEGATE: "Delegate",
    ELICIT_INTENT: "ElicitIntent",
    ELICIT_SLOT: "ElicitSlot",
    NONE: "None",
};
const ConversationMode = {
    AUDIO: "AUDIO",
    TEXT: "TEXT",
};
const InputMode = {
    DTMF: "DTMF",
    SPEECH: "Speech",
    TEXT: "Text",
};
const PlaybackInterruptionReason = {
    DTMF_START_DETECTED: "DTMF_START_DETECTED",
    TEXT_DETECTED: "TEXT_DETECTED",
    VOICE_START_DETECTED: "VOICE_START_DETECTED",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
Object.defineProperty(exports, "LexRuntimeV2ServiceException", {
    enumerable: true,
    get: function () { return LexRuntimeV2ServiceException.LexRuntimeV2ServiceException; }
});
exports.ConfirmationState = ConfirmationState;
exports.ConversationMode = ConversationMode;
exports.DeleteSessionCommand = DeleteSessionCommand;
exports.DialogActionType = DialogActionType;
exports.GetSessionCommand = GetSessionCommand;
exports.InputMode = InputMode;
exports.IntentState = IntentState;
exports.InterpretationSource = InterpretationSource;
exports.LexRuntimeV2 = LexRuntimeV2;
exports.LexRuntimeV2Client = LexRuntimeV2Client;
exports.MessageContentType = MessageContentType;
exports.PlaybackInterruptionReason = PlaybackInterruptionReason;
exports.PutSessionCommand = PutSessionCommand;
exports.RecognizeTextCommand = RecognizeTextCommand;
exports.RecognizeUtteranceCommand = RecognizeUtteranceCommand;
exports.SentimentType = SentimentType;
exports.Shape = Shape;
exports.StartConversationCommand = StartConversationCommand;
exports.StyleType = StyleType;
Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) Object.defineProperty(exports, k, {
        enumerable: true,
        get: function () { return schemas_0[k]; }
    });
});
Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) Object.defineProperty(exports, k, {
        enumerable: true,
        get: function () { return errors[k]; }
    });
});
